src/api/app/models/project.rb

```diff
@@ -1800,6 +1815,30 @@ def self.remove_repositories(repositories, full_remove = false)
     {}
   end
 
+  def save_distributions(distribution_names)
+    distributions = Distribution.all_including_remotes.select{ |distribution| distribution_names.try(:include?, distribution['reponame']) }
+    begin
+      ActiveRecord::Base.transaction do
+        distributions.each do |distribution|
+          repository = repositories.new(name: distribution['reponame'])
+          target_repository =  Repository.find_by_project_and_name(distribution['project'], distribution['repository'])
+          unless target_repository
+            raise ActiveRecord::RecordNotFound
+          end
+          repository.path_elements.new(link: target_repository, position: 1 )
+          distribution['architectures'].each_with_index do |architecture, index|
+            repository.repository_architectures.new(architecture: Architecture.archcache[architecture], position: index + 1)
+          end
+          repository.save!
+        end
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      self.distribution_errors = e.message
+      return false
+    end
+    true
```

Comment:
Instead of having a `return false` at the rescue clause... I would do this like this: 
```ruby
           repository.save!
         end
       end
       true
     rescue ActiveRecord::RecordInvalid => e
       self.distribution_errors = e.message
       false
     end
```